### PR TITLE
tokengen: allow to configure the number of threads

### PR DESCRIPTION
**Pull request by thc202**
Add an options panel, TokenOptionsPanel, to allow to configure the
number of threads used during the token generation.
Change ExtensionTokenGen to hook the TokenParam and TokenOptionsPanel.
Change TokenParam to:
 - Extend VersionedAbstractParam, so that it can be versioned;
 - Rename the property/method from threadPerScan to threadsPerScan;
 - Add constant for the default number of threads and change it from 20
 to 5 (same as other tools, e.g. fuzzer);

Add a button to TokenPanel to allow direct access to the options.
Add help page for the options and update main page to link to it.
Update changes in ZapAddOn.xml file.

### CODE REVIEW

File: src/org/zaproxy/zap/extension/tokengen/resources/Messages.properties [hunk 1]
```diff
@@ -46,6 +46,8 @@ tokengen.label.initialMessage   = <html>To generate and analyse tokens:\n\n<ul><
 tokengen.label.maxEntropy       = Maximum Entropy
 tokengen.panel.mnemonic			= t
 tokengen.panel.title            = Token Gen
+tokengen.optionspanel.name = Token Generator
+tokengen.optionspanel.option.threadsperscan = Number of threads:
 tokengen.results.table.header.timestamp.request = Req. Timestamp
```
Comment (kingthorin) [inline src/org/zaproxy/zap/extension/tokengen/resources/Messages.properties]: Could you make threads have a leading cap? `Threads`

Comment (kingthorin) [reply src/org/zaproxy/zap/extension/tokengen/resources/Messages.properties]: Oops shoulda been on line 50

**Comment by psiinon** [comment]
Looks good but now conflicting

File: src/org/zaproxy/zap/extension/tokengen/resources/Messages.properties (hunk 1 above)
Comment (thc202) [reply src/org/zaproxy/zap/extension/tokengen/resources/Messages.properties]: Changed.

**Comment by psiinon** [comment]
Thanks :)

**Pull request merged**